### DISCUSSION TITLE
Strange behavior of Sigen Daily Grid Energy Export sensor

### DISCUSSION THREAD

**Discussion by Wictorssons** [TypQxQ/Sigenergy-Home-Assistant-Integration#22] [General]
![SigenDailyGridEnergyExport](https://github.com/user-attachments/assets/e94369fc-646e-4af8-a054-d6f598aa6987)
Yesterday the export was zero due to the weather but the sensor Sigen Daily Grid Energy Export reported a jump from zero to more than 50 kWh at 08:00. Is this an error from the rectifier or something with the integration?

**Comment by TypQxQ** [comment]
The daily is calculated from the accumulated sensor that in turn is calculated from sigen_grid_sensor_export_power.
As soon as the Sigen Grid sensor active power returns a negative value, it is assumed that the power flows towards the grid and registers as such for the statistics. 
My theory is that when using more power on one phase from battery. The other phases get offloaded to the grid.
![image](https://github.com/user-attachments/assets/48729349-cb42-4048-8caa-7b9677b06b9d)

**Comment by Wictorssons** [comment]
Even so, isn't the behavior strange? A sudden jump to 50 kWh which is extremely unlikely to have been exported regardless of phase. Or maybe I did not understand your theory. :-)
Here is the month statistics for this.
![SigenDailyGridEnergyExport_Oct](https://github.com/user-attachments/assets/55c738c1-9a7a-4f54-94f7-80df9f353ae3)

**Reply by TypQxQ** [reply]
It is. Check your ”Sigen Grid sensor active power”

**Reply by Wictorssons** [reply]
There is a -6.5 kW spike at 08:00:15.
![SigenGridSensorActivePower_Oct](https://github.com/user-attachments/assets/ae06b45b-9097-4929-aae0-92a090344b25)

**Comment by TypQxQ** [comment]
I updated the update intervals from 10 to 5 seconds in the latest update. See if this makes a difference.
